ocaml/toplevel/genprintval.ml

```diff
@@ -394,6 +394,7 @@ module Make(O : OBJ)(EVP : EVALPATH with type valu = O.t) = struct
                  Oval_constr (Oide_ident (Out_name.create "lazy"), [v])
                end
           | Tconstr(path, ty_list, _) -> begin
```

Comment:
This is technically true for every instance of `tl` (which I assume stands for **t**ype **l**ist), and I'm really not sure it's worth the hassle to change it everywhere when notionally `AppArgs` is still a type list. Here, it is converted to a list on the next line.

Reply:
I think variables named `foo_list` need to have lists in if you don't want to confuse people reading the code. The shadowing on the next line just makes it read extra confusing. Ideally we'd switch all the `tl` cases too, but it's probably not worth it since it is less obviously wrong. That's not a decision based on the hassle of doing it, but rather a question of keeping the diff with upstream smaller.